File: lab2报告/附件/python代码/img2txt.py
import cv2 as cv


img = cv.imread("img02.png")

img_gray = cv.cvtColor(img,cv.COLOR_RGB2GRAY)

img_new = cv.resize(img_gray,(50,50))
size = img_new.shape
cv.imwrite("new_02.jpg",img_new)
with open ("data.txt","w") as f:
    count = 0
    for i in range(50):
        for j in range(50):
            
            if count % 4 == 0 and count != 0:
                f.write("\n")
                print(j)
            f.write((8-len(bin(img_new[i][j])[2:]))*"0"+bin(img_new[i][j])[2:])
            count += 1
            # f.write(bin(img_new[i][j])[2:]+"\n")

    f.close()

File: lab2/img2txt.m
im = imread('img02.png');
im1 = rgb2gray(im);
im1 = imresize(im1, [50 50]);
fi = fopen('data.txt', 'wt');
[m, n] =size(im1);
for i=1:1:m
    for j=1:1:n
        if j==n
            fprintf(fi, '%x\n', im1(i,j));
        else
            fprintf(fi, '%x\n', im1(i,j));
        end
    end
end
fclose(fi);

